cypress/integration/auth.spec.js

```diff
@@ -0,0 +1,7 @@
+describe("Authentication", () => {
+  it("has a login button", () => {
+    cy.visit("http://localhost:3000");
+
+    cy.get("[data-cy=login]").should("exist");
```

Comment:
This is definitely one way to find things on a page, but cypress has a lot of unique tools to help with this.

So there are generally speaking 2 ways of identifying elements in cypress: 

1. `cy.get` which looks for DOM elements by tags/attributes
2. `cy.contains` which looks for DOM elements containing some text

There is another function `cy.within` which allows you to constrict the search to everything under a certain node in the DOM.

Something that might be worth discussing is which of these styles/mixes of styles we want to use and when we should use them.

Reply:
So that's code directly from Bryan's initial example.

@scottpchow23 are you commenting just to spark discussion, or is there a change you'd like to see here?

Reply:
This is just to spark a discussion, nothing more.

Reply:
This approach is documented as a "best practice" here: https://docs.cypress.io/guides/references/best-practices.html#Selecting-Elements

Reply:
True, and it works when you're looking for one unique element on a page. I think I've run into trouble with that when I'm testing specific table data/verifying certain data is being displayed on the page.

Reply:
Either I think it's worth discussing; I'm just giving my perspective as someone who wrote cypress tests for a summer.